RELEASING.md

```diff
@@ -12,7 +12,7 @@
 - On GitHub create a new release for the tag and take note of any relevant changes.
   - Include a brief summary of all the changes
   - Include links to the Pull Requests that introduced these changes
-- Update the [documentation website](https://shopify.github.io/themekit/)
+- Update the [documentation website](https://shopify.dev/tools/theme-kit)
   - run `make serve_docs`
```

Comment:
As we're moving the docs to shopify.dev then `make serve_docs` would no longer apply (and we should probably remove this script from the repo).

Reply:
Also, we should think about the workflow for how doc changes are deployed alongside with new releases. Perhaps the first step should be to open the docs PR on shopify-dev, and get it approved. Then once the new Theme Kit version is released, the docs PR can be merged and deployed right away.